bin/critterget.py

```diff
@@ -520,6 +531,131 @@ def getGenericErrorMon(appId, appName,graph,groupby,messagetype):
 
 """--------------------------------------------------------------"""
 
+def getUserflowsSummary(app_id, app_name, message_type):
+    """Calls the transactions summary endpoint for an app
+
+    :param app_id: (string) App ID used to request data from the API
+    :param app_name: (string) Human-readable app name
+    :param message_type: (string) Type of message (for sorting in Splunk)
+    :return: None
+    """
+    uri = 'transactions/{}/summary/'.format(app_id)
+
+    response = apicall(uri)
+
+    try:
+        messages = u','.join([u'("{}",{},{})'.format(metric, data[VALUE], data[CHANGE_PCT]) for metric, data in
+                              response[SERIES].iteritems()])
+        print u'{} MessageType={} appName="{}" appId="{}" DATA {}'.format(
+            DATETIME_OF_RUN, message_type, app_name, app_id, messages)
+    except KeyError as e:
+        print (u'{} MessageType="ApteligentError" Error: Could not access {} '
+               u'in {}.'.format(DATETIME_OF_RUN, str(e), message_type))
+        return None, None
+
+def getUserflowsRanked(app_id, app_name, category, message_type):
+    """Calls the transactions ranked endpoint to get the top failed transactions
+
+    :param app_id: (string) App ID used to request data from the API
+    :param app_name: (string) Human-readable app name
+    :param category: (string) Kind of transaction to return (e.g. failed, succeeded)
+    :param message_type: (string) Type of message (for sorting in Splunk), equivalent to category
+    :return: None
+    """
+
+    uri = 'transactions/{}/ranked/{}/'.format(app_id, category)
+
+    response = apicall(uri)
+
+    try:
+        messages = u','.join([u'("{}",{},{})'.format(group[NAME], group[FAILURE_RATE], group[UNIT][TYPE]) for group in
+                              response[GROUPS]])
+        print u'{} MessageType={} appName="{}" appId="{}"  DATA {}'.format(
+            DATETIME_OF_RUN, message_type, app_name, app_id, messages)
+    except KeyError as e:
+        print (u'{} MessageType="ApteligentError" Error: Could not access {} '
+               u'in {}.'.format(DATETIME_OF_RUN, str(e), message_type))
+
+
+def getUserflowsDetails(app_id, app_name):
+    """Call the userflows details/change endpoint for an app
+
+    :param app_id: (string) App ID used to request data from the API
+    :param app_name: (string) Human-readable app name
+    :return: None
+    """
+    uri = 'transactions/{}/details/change/P1M?pageNum=1&pageSize=10&sortBy=name&sortOrder=ascending'.format(app_id)
```

Comment:
This is the query that is used to populate the equivalent table on the portal. I want to make sure the data match as much as possible.